src/geometry.rs

```diff
@@ -19,20 +19,50 @@ pub struct Shape {
 }
 
 impl Shape {
-    /// Create an empty shape.
+    /// Creates an empty shape.
     pub fn empty() -> Self {
```

Comment:
Hmm... do we really need that method? One can use `Shape::default` now

Reply:
I'm inclined to agree. I kept this because there's a note in the Rust API guidelines about constructors calling `Default::default`. See [the bottom of this page](https://rust-lang-nursery.github.io/api-guidelines/predictability.html#constructors-are-static-inherent-methods-c-ctor).

Reply:
> "Note that it is common and expected for types to implement both Default and an empty new constructor. new is the constructor convention in Rust, and users expect it to exist, so if it is reasonable for the basic constructor to take no arguments, then it should, even if it is functionally identical to default."

Perhaps we should rename it as `new` instead?

Reply:
> Some types' constructors, most notably I/O resource types, use distinct naming conventions for their constructors, as in File::open, Mmap::open, TcpStream::connect, and UpdSocket::bind. In these cases names are chosen as appropriate for the domain.

Actually, `empty` is more appropriate for our case.

Reply:
Well, let's keep `empty` as it is